src/charm.py

```diff
@@ -452,7 +466,31 @@ def _prometheus_config(self) -> str:
             "metrics_path": "/metrics",
             "honor_timestamps": True,
             "scheme": "http",
-            "static_configs": [{"targets": [f"localhost:{self._port}"]}],
+            "static_configs": [
+                {
+                    "targets": [f"localhost:{self._port}"],
+                    "labels": {
```

Comment:
`host` is not (`__address__`, `__name__`, `instance`, and some others are).

The usage here is for the integration tests in `scrape_multiunit`. It could be anything. For anyone querying, I suppose that `host: "localhost"` is as meaningful as anything, and definitely more meaningful than `type: "whatever"`, since `type:` is not a commonly used label for any reason.

The label could be `abcdefghijklmnopqrstuvwxyz`. It doesn't matter. It's just a selector for promql.

Reply:
Should we fix our itests then instead of introducing a dummy label for everyone?

Reply:
I don't understand the hangup around this particular label. It is not there just for tests, and it is not a "dummy" label.

It's additional cardinality (easily visible in the Prometheus query UI) which allows users to say "only show me metrics from _this_ Prometheus" without needing to remember/keep track of topology labels.

It may be non-standard, but there are plenty of non-standard labels for applications, and the question is basically "for a user interacting with a query language who may want to add additional specificity, is it more intuitive to look for `localhost` or to divine that `type: self` actually means `localhost`?"